pvlib/modelchain.py

```diff
@@ -926,17 +952,65 @@ def prepare_inputs(self, weather, times=None):
             self.weather['temp_air'] = 20
         return self
 
+    def prepare_inputs_from_poa(self, total_irrad):
```

Comment:
as discussed below, maybe stick with `weather` and the input argument. Note the `if` statement at the end of the method should fail because `weather` is not defined.